### PR TITLE
gh-114058: Improve method information in redundancy eliminator

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
* Issue: gh-114058

### CODE REVIEW

File: Python/tier2_redundancy_eliminator_bytecodes.c [hunk 1]
```diff
@@ -340,7 +344,7 @@ dummy_func(void) {
 
         assert(self_or_null != NULL);
         assert(args != NULL);
-        if (sym_is_not_null(self_or_null)) {
+        if (!sym_is_null(self_or_null)) {
```
Comment (markshannon) [inline Python/tier2_redundancy_eliminator_bytecodes.c]: This is not the same for top.

File: Python/optimizer_analysis.c [hunk 2]
```diff
@@ -259,12 +259,6 @@ sym_is_known(_Py_UOpsSymType *sym)
     return sym_has_flag(sym, KNOWN);
 }
 
-static inline bool
-sym_is_not_null(_Py_UOpsSymType *sym)
```
Comment (markshannon) [inline Python/optimizer_analysis.c]: Why are you removing this?

**Review by markshannon — CHANGES_REQUESTED** [review]
I don't think you can remove `sym_is_not_null`

**Comment by Fidget-Spinner** [comment]
I have made the requested changes; please review again

**Review by markshannon — APPROVED** [review]
Looks good

**Pull request merged**